### PR TITLE
Update user map for June 2017 #7696

### LINKED ISSUE #7696: Update user map for June 2017
Follow #7215 and similar.
New data to consider:

Country|Institute
-- | --
Australia | Carey Baptist College
Australia | Monash University
Australia | University of Canberra
Australia | University of South Australia
Australia | University of Western Australia
Belgium | KU Leuven
Brazil | AEDB
Brazil | The State University of Londrina
Brazil | UNESP
Brazil | Universidade Católica de Brasília
Brazil | Universidade de São Paulo
Canada | College de Bois-de-Boulogne
Canada | St. Barbara Catholic Elementary School
Colombia | Universidad del Norte
Colombia | Universidad del Norte
Costa Rica | Colegio Calasanz Hispanocostarricense
Estonia | Oneself Limited
Estonia | Tartu University Pärnu college
France | EMLYON Business School
France | IESEG School of Management
Germany | TU Munich
India | IIIT, Allahabad
India | Payilagam
Malaysia | International University of Malaya-Wales
Malaysia | Universiti Putra Malaysia
Mexico | CETYS Universidad
Mexico | Lemon Three
Mexico | UDLAP
Nederlands | Hogeschool van Amsterdam
Netherlands | Eindhoven University of Technology
Netherlands | University of Twente
New Zealand | Eastern Institute of Technology
New Zealand | Massey University
New Zealand | Massey University
New Zealand | Queen's High School
New Zealand | The University of Auckland
Philippines | Oneself Limited
Portugal | School of education Porto
Russia | Higher School of Economics
Singapore | MCI Group
Singapore | SHATEC Institutes
Singapore | Singapore Institute of Technology
Singapore | Singapore Polytechnic
Singapore | St. Joseph's Institution
South Korea | SUNY Korea
Spain | Cooke e-learning
Spain | Universidad de Barcelona
Spain | University of Barcelona
Switzerland | Pädagogische Hochschule St. Gallen
Taiwan | Chung-Yuan Christian University
Turkey | Kredi Akademi
UK | Brunel University
UK | Hult International Business School
UK | University College London
USA | Arizona State University
USA | Carnegie Mellon University
USA | College America
USA | Colorado State University
USA | Delaware Technical Community College
USA | Flagship Computing
USA | Grand Valley State University
USA | Hult Business School
USA | Illinois State University
USA | Johns Hopkins University
USA | Kennesaw State University
USA | Medical College of Wisconsin
USA | Medical College of Wisconsin
USA | Morgan County Charter School System
USA | Morgan County Middle School
USA | NC State University
USA | NC State University
USA | Nichols College
USA | Noncommsioned Officer Academy at the USAJFKSWCS
USA | Northeastern University
USA | NYU Stern
USA | Phillip & Sala Burton Academic High School
USA | Portland State University
USA | Regis University
USA | Regis University
USA | Southern Illinois University Edwardsville
USA | UNC - Chapel Hill
USA | University at Albany, SUNY
USA | University of California, Irvine
USA | University of Dallas
USA | University of Georgia
USA | University of New Hampshire

### PR DESCRIPTION
**Pull request by davel37**
Fixes #7696

Creating the pull request now. I can complete the changes that are required on Sunday.

### CODE REVIEW

**Comment by davel37** [comment]
Ready for review.

All entries have been added and this issue is ready for review. (Although I am slightly puzzled about this conficting file message). Having studied how this issue was resolved in previous months it appears the only changes made were to the useMapData.json file.

**Comment by davel37** [comment]
Ready for review.

File: src/test/java/teammates/test/cases/browsertests/InstructorSubmissionAdjustmentUiTest.java [hunk 1]
```diff
@@ -39,8 +39,8 @@ protected void prepareTestData() {
         removeAndRestoreDataBundle(testData);
     }
 
-    @Test
-    public void testAdjustmentOfSubmissions() {
+
+    public void testAdjustmentOfSubsmissions() {
```
Comment (whipermr5) [inline src/test/java/teammates/test/cases/browsertests/InstructorSubmissionAdjustmentUiTest.java]: This is an unrelated change that also happens to be causing the build failure (take a look at the build log by clicking on the red `x` next to your last commit).

Comment (davel37) [inline src/test/java/teammates/test/cases/browsertests/InstructorSubmissionAdjustmentUiTest.java]: Ok I'll look into it. I'm a bit unclear as to what changes I need to make to this InstructorSubmissionsAdjustmentUiTest.java code.

Comment (whipermr5) [inline src/test/java/teammates/test/cases/browsertests/InstructorSubmissionAdjustmentUiTest.java]: None at all :) Any unintentional change should be reverted.

File: src/test/java/teammates/test/cases/browsertests/InstructorSubmissionAdjustmentUiTest.java [hunk 2]
```diff
@@ -39,9 +39,12 @@ protected void prepareTestData() {
         removeAndRestoreDataBundle(testData);
     }
 
+
+
```
Comment (whipermr5) [inline src/test/java/teammates/test/cases/browsertests/InstructorSubmissionAdjustmentUiTest.java]: This file is still showing up in the changeset due to 3 added blank lines.

File: src/main/webapp/data/userMapData.json [hunk 3]
```diff
@@ -253,8 +256,12 @@
       "Universidad de Cundinamarca",
       "Universidad de Los Andes",
       "Universidad de San Buenaventura Bogotá",
+      "Universidad del Norte",
       "Universidad El Bosque"
     ],
+    "Costa Rica": [
+    "Colegio Calasanz Hispanocostarricense"
```
Comment (whipermr5) [inline src/main/webapp/data/userMapData.json]: This needs indentation.

File: src/main/webapp/data/userMapData.json [hunk 4]
```diff
@@ -286,7 +293,10 @@
       "The International School of Elite Education"
     ],
     "Estonia": [
-      "University of Tartu"
+      "Oneself Limited",
+      "University of Tartu",
+      "Tartu University Pärnu college"
+
```
Comment (whipermr5) [inline src/main/webapp/data/userMapData.json]: Delete this blank line.

File: src/main/webapp/data/userMapData.json [hunk 5]
```diff
@@ -1489,10 +1529,12 @@
       "Truman State University",
       "Tufts University",
       "UC San Diego",
+      "UNC - Chapel Hill",
```
Comment (whipermr5) [inline src/main/webapp/data/userMapData.json]: This should be moved to after the next line to maintain alphabetical order. Also, the dash can be dropped for consistency with `UNC Pembroke`.

**Comment by whipermr5** [comment]
@davel37 The [`Files changed`](https://github.com/TEAMMATES/teammates/pull/7703/files) tab still shows 2 files (it should only show 1).

File: src/main/webapp/data/userMapData.json [hunk 6]
```diff
@@ -596,9 +612,13 @@
     ],
     "New Zealand": [
       "Auckland University of Technology",
+      "Eastern Institute of Technology",
       "Manukau Institute of Technology",
+      "Massey University",
       "Media Design School",
       "Otago Polytechnic",
+      "Queen's High School",
+      "The University of Auckland",
```
Comment (whipermr5) [inline src/main/webapp/data/userMapData.json]: `University of Auckland` already exists below.

File: src/main/webapp/data/userMapData.json [hunk 7]
```diff
@@ -387,6 +397,7 @@
       "Gujarat University",
       "Guru Gobind Singh Indraprastha University",
       "Ideas That Scale Accelerator LLP",
+      "IIIT, Allahabad",
```
Comment (whipermr5) [inline src/main/webapp/data/userMapData.json]: Drop the comma for consistency with `IIIT Delhi`.

File: src/test/java/teammates/test/cases/browsertests/InstructorSubmissionAdjustmentUiTest.java [hunk 8]
```diff
@@ -38,7 +38,7 @@ protected void prepareTestData() {
 
         removeAndRestoreDataBundle(testData);
     }
-
+
```
Comment (whipermr5) [inline src/test/java/teammates/test/cases/browsertests/InstructorSubmissionAdjustmentUiTest.java]: Almost there!

Comment (whipermr5) [inline src/test/java/teammates/test/cases/browsertests/InstructorSubmissionAdjustmentUiTest.java]: This is also causing the build failure as indicated by the red `x` next to your commits.

File: src/test/java/teammates/test/cases/browsertests/InstructorSubmissionAdjustmentUiTest.java [hunk 9]
```diff
@@ -38,7 +38,6 @@ protected void prepareTestData() {
 
         removeAndRestoreDataBundle(testData);
     }
-
```
Comment (whipermr5) [inline src/test/java/teammates/test/cases/browsertests/InstructorSubmissionAdjustmentUiTest.java]: There originally was a blank line here (just that it didn't include trailing spaces).

File: src/test/java/teammates/test/cases/browsertests/InstructorSubmissionAdjustmentUiTest.java (hunk 8 above)
Comment (whipermr5) [inline src/test/java/teammates/test/cases/browsertests/InstructorSubmissionAdjustmentUiTest.java]: Once these 4 trailing spaces are deleted, this entire file should disappear from the changeset.

File: src/main/webapp/data/userMapData.json [hunk 10]
```diff
@@ -162,6 +163,7 @@
       "Cegep de Bois-de-Boulogne",
       "Chetwynd Secondary School",
       "Collège Ahuntsic",
+      "College de Bois-de-Boulogne",
```
Comment (whipermr5) [inline src/main/webapp/data/userMapData.json]: This looks similar to `Cegep de Bois-de-Boulogne` 3 lines above; might it be the same institution?

File: src/main/webapp/data/userMapData.json [hunk 11]
```diff
@@ -203,6 +205,7 @@
       "Simon Fraser University",
       "Southern Alberta Institute of Technology",
       "St. Andrew's College",
+      "St. Barbara Catholic Elementary School",
```
Comment (whipermr5) [inline src/main/webapp/data/userMapData.json]: Drop the `School` suffix for consistency with `St. David Catholic Elementary` below?

File: src/main/webapp/data/userMapData.json [hunk 12]
```diff
@@ -286,7 +293,9 @@
       "The International School of Elite Education"
     ],
     "Estonia": [
-      "University of Tartu"
+      "Oneself Limited",
+      "University of Tartu",
+      "Tartu University Pärnu college"
```
Comment (whipermr5) [inline src/main/webapp/data/userMapData.json]: Capitalise `college`?

File: src/main/webapp/data/userMapData.json [hunk 13]
```diff
@@ -660,6 +680,7 @@
       "Escola Profissional de Almada",
       "Faculdade de Engenharia da Universidade do Porto",
       "Musaic",
+      "School of education Porto",
```
Comment (whipermr5) [inline src/main/webapp/data/userMapData.json]: Capitalise `education`?

File: src/main/webapp/data/userMapData.json [hunk 14]
```diff
@@ -725,6 +747,7 @@
       "Rosyth School",
       "Sengkang Secondary School",
       "Serangoon Junior College",
+      "SHATEC Institutes",
```
Comment (whipermr5) [inline src/main/webapp/data/userMapData.json]: I think the full name is `SHATEC Institutes Pte Ltd`, which is the same as the one below.

File: src/main/webapp/data/userMapData.json [hunk 15]
```diff
@@ -1328,6 +1364,7 @@
       "Northwood University",
       "Norwich Technical High School",
       "Nyack High School",
+      "NYU Stern",
```
Comment (whipermr5) [inline src/main/webapp/data/userMapData.json]: This seems to be a business school within `New York University`, which is already in the list.

File: src/main/webapp/data/userMapData.json [hunk 16]
```diff
@@ -1490,9 +1528,11 @@
       "Tufts University",
       "UC San Diego",
       "UM Flint",
+      "UNC Chapel Hill",
```
Comment (whipermr5) [inline src/main/webapp/data/userMapData.json]: `University of North Carolina at Chapel Hill` is already in the list

File: src/main/webapp/data/userMapData.json [hunk 17]
```diff
@@ -1490,9 +1528,11 @@
...
       "UNC Pembroke",
       "Union Mine High School",
       "United States Military Academy Preparatory School",
+      "University at Albany, SUNY",
```
Comment (whipermr5) [inline src/main/webapp/data/userMapData.json]: This is already in the list, two lines down.

File: src/main/webapp/data/userMapData.json [hunk 18]
```diff
@@ -335,6 +344,7 @@
       "SHIFTSCHOOL",
       "Technical University of Munich ",
       "TU Dresden",
+      "TU Munich",
```
Comment (whipermr5) [inline src/main/webapp/data/userMapData.json]: Same as `Technical University of Munich` two lines above?

File: src/main/webapp/data/userMapData.json (hunk 10 above)
Comment (davel37) [inline src/main/webapp/data/userMapData.json]: Yes,  it appears to be. I'll remove the other entry as Collège de Bois-de-Boulogne, seems to be the standard name for it.

File: src/main/webapp/data/userMapData.json (hunk 11 above)
Comment (whipermr5) [inline src/main/webapp/data/userMapData.json]: Either that or add `School` to the below.

File: src/main/webapp/data/userMapData.json (hunk 16 above)
Comment (whipermr5) [inline src/main/webapp/data/userMapData.json]: Pending removal of this.

Comment (davel37) [inline src/main/webapp/data/userMapData.json]: Done!

File: src/main/webapp/data/userMapData.json [hunk 19]
```diff
@@ -1,5 +1,5 @@
 {
-  "lastUpdated": "31 May 2017",
+  "lastUpdated": "9 July 2017",
```
Comment (damithc) [inline src/main/webapp/data/userMapData.json]: This should be `30 June`. To minimize future errors, we can in fact use `End of June 2017`

File: src/main/webapp/data/userMapData.json [hunk 20]
```diff
@@ -286,7 +292,9 @@
       "The International School of Elite Education"
     ],
     "Estonia": [
-      "University of Tartu"
+      "Oneself Limited",
```
Comment (damithc) [inline src/main/webapp/data/userMapData.json]: this looks suspicious. Let's remove.

File: src/main/webapp/data/userMapData.json [hunk 21]
```diff
@@ -644,6 +661,7 @@
       "Miriam College High School",
       "Misamis Oriental Institute Of Science and Technology",
       "New Era University",
+      "Oneself Limited",
```
Comment (damithc) [inline src/main/webapp/data/userMapData.json]: Here it is again. Let's remove this one too.

Comment (davel37) [inline src/main/webapp/data/userMapData.json]: Ok!

**Comment by davel37** [comment]
@whipermr5 Thanks for your help. I certainly learned a lot with this pull request. Onwards and upwards ! :smile:

**Pull request merged**